scripts/generate-javadoc-plugins.groovy

```diff
@@ -1,90 +1,35 @@
-
-    // we obtain the GAV value, and tokenize it at the ":"
-    def gav = value.gav.split(":");
-
-    // For the GroupID we need to replace the . with a / which is what we search for in the URL
-    gid = gav[0].replace(".", "/")
-
-    // Get the artifactID and version number as well.
-    aid = gav[1]
-    ver = gav[2]
-
-    // The plugin location is defined as:
-    // "https://repo.jenkins-ci.org/releases/groupWithSlashes/artifactId/version/artifactId-version-javadoc.jar"
-    def pluginLoc  = pluginLocation + gid + "/" + aid + "/" + ver + "/" + aid + "-" + ver + "-javadoc.jar";
-
-    // Define the directory as to where the plugin should be extracted to
-    def plugin_dir="build/site/plugin/"+ aid + "/"
-
-    // Create the directory where the plugin should exist.
-    new File(plugin_dir).mkdirs();
-
-    // Create the new jar file
-    def file = new File(plugin_dir, "/" + aid + ".jar");
-
-    // We need an output stream to write the content from the url to the file.
-    def fos = file.newOutputStream()
-
-    def name = value.title
-    def id = value.name
-    def version = value.version
+Set<String> pluginsAIDs = null;
+if (args.length > 0 && !args[0].trim().empty) {
+    if (pluginsAIDs == null) {
+        pluginsAIDs = new HashSet<>(args.length)
+    }
 
-    try {
-        // Write the contents of the *-javadoc.jar to the file
-        file << new URL(pluginLoc).openStream()
+    def argPluginIDs = args[0].trim().split("[\\s,]+")
+    for (def pluginID : argPluginIDs) {
+        pluginsAIDs.add(pluginID)
+    }
+    println "Plugins to be published: ${argPluginIDs.join(",")}"
+}
 
-        // Unzip the contents to the plugin directory
-        ant.unzip(
-                src:file,
-                dest:plugin_dir,
-                overwrite:true)
+// Start building the plugin group
+def indexBuilder = new JavadocGroupBuilder("plugin", "plugin", "Jenkins Plugins Javadoc", pluginsAIDs);
 
-        indexHtml << "<div id='${id}'><h2><a href='${id}'>${name}</a><span class='version'>${version}</span></h2><p><tt>${id}</tt></p><p><a href='${id}'>Javadoc</a></p><p><a href='https://plugins.jenkins.io/${id}'>Plugin Information</a></p></div>"
-        jsonUrlMap[id] = [url: baseUrl + id]
-    } catch (FileNotFoundException e) {
+// For each plugin located in the update center
 
-        // This will only be encountered if there is no javadocs in our repo. We can safely move on.
-        println "No javadoc found for: " + aid
-        indexHtml << "<div id='${id}' class='missing'><h2>${name}<span class='version'>${version}</span></h2><p><tt>${id}</tt></p><p>No Javadoc has been published for this plugin.</p><p><a href='https://plugins.jenkins.io/${id}'>Plugin Information</a></p></div>"
-    } finally {
-        fos.close();
-    }
+String location = "http://updates.jenkins.io/current/update-center.actual.json"
+def json = new JsonSlurper().parseText(new URL (location).text);
```

Comment:
It's just a stylistic thing, but the alternation between static typing and dynamic typing is jarring to read.